src/gui/qgsmapcanvas.cpp

```diff
@@ -1246,6 +1246,15 @@ void QgsMapCanvas::updateDevicePixelFromScreen()
   refresh();
 }
 
+void QgsMapCanvas::onMapShadingChanged()
+{
+  if ( !mProject )
+    return;
+  mSettings.setShadingRenderer( mProject->mapShadingRenderer() );
+  mCache->clear();
```

Comment:
clearing the whole cache is maybe a bit harsh? :-)  the underlying color image stay the same, and the elevation map images also stay the same, we only need to re-apply shading. Not clearing the cache would allow much better UX when adjusting parameters, as one would not need to wait until all data get fetched again...